src/components/UI/Content/index.tsx

```diff
@@ -1,36 +1,50 @@
 import { faChevronDown } from "@fortawesome/free-solid-svg-icons";
 import { FontAwesomeIcon } from "@fortawesome/react-fontawesome";
-import React, { PropsWithChildren } from "react";
+import React, { PropsWithChildren, useRef, useState } from "react";
 import constants from "../../../constants/UI";
 
 import utilStyles from "../../../styles/utils.module.css";
 import styles from "./Content.module.css";
 
 const Content: React.FC<PropsWithChildren<any>> = (props) => {
+  const [isActive, setIsActive] = useState(true);
+  const contentEl = useRef<HTMLDivElement>(null);
   return (
     <>
       <hr className={styles.line} />
       <div className={styles.accordian}>
-        <input
-          type="checkbox"
-          name="accordian"
-          id={props.title}
-          className={styles.accordianInput}
-          defaultChecked
-        />
-        <label htmlFor={props.title} className={styles.accordianLabel}>
-          <div className={styles.contentHeader}>
-            {" "}
-            <h1 className={utilStyles.contentTitle}>{props.title} </h1>
-            <FontAwesomeIcon
-              icon={faChevronDown}
-              size="sm"
-              color={constants.contentTitleColor}
-              className={styles.chevronDown}
-            />
-          </div>
-        </label>
-        <div className={styles.content}>{props.children}</div>
+        <div
+          className={styles.contentHeader}
+          onClick={() => {
+            setIsActive(!isActive);
+            console.log(isActive);
+          }}
+        >
+          {" "}
+          <h1 className={utilStyles.contentTitle}>{props.title} </h1>
+          <FontAwesomeIcon
+            icon={faChevronDown}
+            size="sm"
+            color={constants.contentTitleColor}
+            className={`${styles.chevronDown} ${
+              isActive ? styles.rotated : ""
+            }`}
+          />
+        </div>
+
+        <div
+          ref={contentEl}
+          className={styles.content}
+          style={{
+            height: `${
+              isActive
+                ? contentEl.current?.scrollHeight.toString() + "px"
```

Comment:
issue: on the first reload, height is set as `undefinedpx` , we should avoid those cases.